api/tests/integration/domain/usecases/create-pro-organization-with-tags_test.js

```diff
@@ -199,145 +391,176 @@ describe('Integration | UseCases | create-pro-organization', function () {
+
+    const organizationsWithTagsAlreadyExist = [
       {
         externalId: 'b200',
         name: 'Youness et Fils',
         provinceCode: '123',
         credit: 0,
         email: 'youness@example.net',
         locale: 'fr-fr',
-        tags: 'TagNotFound',
+        tags: 'Tag1_Tag2',
+        createdBy: userId,
+        documentationUrl: 'http://www.pix.fr',
+        organizationInvitationRole: 'ADMIN',
       },
       {
-        externalId: 'b201',
-        name: '',
+        externalId: 'b300',
+        name: 'Andreia & Co',
         provinceCode: '345',
         credit: 10,
         email: 'andreia@example.net',
         locale: 'fr-fr',
-        tags: 'TagNotFound',
+        tags: 'Tag2_Tag3',
+        createdBy: userId,
+        documentationUrl: 'http://www.pix.fr',
+        organizationInvitationRole: 'ADMIN',
       },
       {
-        externalId: 'b202',
+        externalId: 'b400',
         name: 'Mathieu Bâtiment',
         provinceCode: '567',
         credit: 20,
         email: 'mathieu@example.net',
         locale: 'fr-fr',
-        tags: 'TagNotFound',
+        tags: 'Tag1_Tag3',
+        createdBy: userId,
+        documentationUrl: 'http://www.pix.fr',
+        organizationInvitationRole: 'ADMIN',
       },
     ];
 
     // when
-    const error = await catchErr(createProOrganizationsWithTags)({
+    await createProOrganizationsWithTags({
       domainTransaction,
-      organizations: organizationsWithTagsWithOneMissingName,
+      organizations: organizationsWithTagsAlreadyExist,
       organizationRepository,
       tagRepository,
       organizationTagRepository,
       organizationInvitationRepository,
     });
 
     // then
-    expect(error).to.be.instanceOf(ObjectValidationError);
-    expect(error.message).to.be.equal('Le nom de l’organisation n’est pas présent.');
     const organizationsInDB = await knex('organizations').select();
-    expect(organizationsInDB.length).to.equal(0);
+    expect(organizationsInDB.length).to.equal(3);
     const organizationTagsInDB = await knex('organization-tags').select();
-    expect(organizationTagsInDB.length).to.equal(0);
+    expect(organizationTagsInDB.length).to.equal(6);
+
+    for (const organization of organizationsWithTagsAlreadyExist) {
+      const organizationInDB = await knex('organizations')
+        .first('id', 'externalId', 'name', 'provinceCode', 'credit', 'email')
+        .where({ externalId: organization.externalId });
+      expect(omit(organizationInDB, 'id')).to.be.deep.equal(
+        omit(organization, 'locale', 'tags', 'type', 'createdBy', 'documentationUrl', 'organizationInvitationRole')
+      );
+
+      const organizationTagInDB = await knex('organization-tags')
+        .select()
+        .where({ organizationId: organizationInDB.id });
+      expect(organizationTagInDB.length).to.equal(2);
+
+      organizationTagInDB.forEach((value) => {
+        expect(value.organizationId).to.be.equal(organizationInDB.id);
+      });
+    }
```

Comment:
**suggestion**
En fait je pense comprendre pourquoi j'ai eu tant de mal à lire ce test : est-ce qu'il ne testerait pas 2 choses en même temps : la création des orga avec les bons attributs + la création de leur tag associé ? 
Si c'est ça on pourrait peut être se séparer ce test en deux pour éviter d'avoir à expect des choses sur l'orga et des choses sur l'organization-tag en même temps (et aussi s'évite la boucle for qui me semble assez difficile à lire)

Reply:
oui le code crée les deux en même temps dans le script et ça rollback si souci 
 
` await domainTransaction.execute(async (domainTransaction) => {
    const organizationsToCreate = Array.from(organizationsData.values()).map((data) => data.organization);

    createdOrganizations = await organizationRepository.batchCreateProOrganizations(
      organizationsToCreate,
      domainTransaction
    );

    const organizationsTags = createdOrganizations.flatMap(({ id, externalId, name }) => {
      return organizationsData.get(externalId).tags.map((tagName) => {
        const foundTag = allTags.find((tagInDB) => tagInDB.name === tagName.toUpperCase());
        if (foundTag) {
          return new OrganizationTag({ organizationId: id, tagId: foundTag.id });
        } else {
          throw new OrganizationTagNotFound(`Le tag ${tagName} de l'organisation ${name} n'existe pas.`);
        }
      });
    });

    await organizationTagRepository.batchCreate(organizationsTags, domainTransaction);
  });`

Reply:
yess, du coup j'aurai plus vu : 1 test pour vérifier que `await organizationRepository.batchCreateProOrganizations` fasse son taff et 1 autre test pour vérifier que `await organizationTagRepository.batchCreate` fasse son taff. 
Mais c'est du chipotage , on peut merge la PR en l'état (elle est devenue assez grosse comme ça pour pas forcément tout refacto non plus 😅 )

Reply:
oui, d'accord sur le principe de bien montrer que ça fait deux choses et d'accord pour ne pas aller trop loin dans le refacto qui était en bonus sur ce ticket ^^